src/main/java/io/strimzi/kafka/quotas/StaticQuotaCallback.java

```diff
@@ -59,15 +62,34 @@ public Double quotaLimit(ClientQuotaType quotaType, Map<String, String> metricTa
         if (ClientQuotaType.PRODUCE.equals(quotaType) && currentStorageUsage > storageQuotaSoft && currentStorageUsage < storageQuotaHard) {
             double minThrottle = quotaMap.getOrDefault(quotaType, Quota.upperBound(Double.MAX_VALUE)).bound();
             double limit = minThrottle * (1.0 - (1.0 * (currentStorageUsage - storageQuotaSoft) / (storageQuotaHard - storageQuotaSoft)));
-            log.debug("Throttling producer rate because disk is beyond soft limit. Used: {}. Quota: {}", storageUsed, limit);
+            maybeLog(lastLoggedMessageSoftTimeMs, "Throttling producer rate because disk is beyond soft limit. Used: {}. Quota: {}", storageUsed, limit);
             return limit;
         } else if (ClientQuotaType.PRODUCE.equals(quotaType) && currentStorageUsage >= storageQuotaHard) {
-            log.debug("Limiting producer rate because disk is full. Used: {}. Limit: {}", storageUsed, storageQuotaHard);
+            maybeLog(lastLoggedMessageHardTimeMs, "Limiting producer rate because disk is full. Used: {}. Limit: {}", storageUsed, storageQuotaHard);
             return 1.0;
         }
         return quotaMap.getOrDefault(quotaType, Quota.upperBound(Double.MAX_VALUE)).bound();
     }
 
+    /**
+     * Put a small delay between logging
+     */
+    private void maybeLog(AtomicLong lastLoggedMessageTimeMs, String format, Object... args) {
+        long now = System.currentTimeMillis();
+        final boolean[] shouldLog = {true};
+        lastLoggedMessageTimeMs.getAndUpdate(current -> {
+            if (now - current >= LOGGING_DELAY_MS) {
+                shouldLog[0] = true;
+                return now;
+            }
+            shouldLog[0] = false;
+            return current;
+        });
+        if (shouldLog[0]) {
+            log.debug(format, args);
+        }
```

Comment:
Actually, one more thing: We can wrap this whole thing in an `if (log.isDebugEnabled())`